sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -715,131 +716,254 @@ class GroupedAggregations extends Arm {
+      isRunningBatched: Boolean,
       inputCb: ColumnarBatch,
-      outputColumns: Array[ColumnVector]): Unit = {
+      outputColumns: Array[cudf.ColumnVector]): Unit = {
     runningWindowOptimizedData.foreach {
       case (func, outputIndexes) =>
-        val agg = func.scanAggregation
-        val replace = func.scanReplaceNulls
-        require(func.boundInputLocations.length == 1)
-        val inputColIndex = func.boundInputLocations.head
-        val inputCol = inputCb.column(inputColIndex).asInstanceOf[GpuColumnVector].getBase
-        val replaced = withResource(inputCol.scan(agg, ScanType.INCLUSIVE, NullPolicy.EXCLUDE)) {
-          scanned =>
-            // For scans when nulls are excluded then each input row that has a null in it the
-            // output row also has a null in it. Typically this is not what we want, because
-            // for windows that only happens if the first values are nulls. So we will then call
-            // replace nulls as needed to fix that up. Typically the replacement policy is
-            // preceding.
-            if (replace.isDefined) {
-              scanned.replaceNulls(replace.get)
-            } else {
-              scanned.incRefCount()
-            }
+        val aggAndReplaces = func.scan(isRunningBatched)
+        // For now we need at least one column. For row number in the future we might be able
+        // to change that, but I think this is fine.
+        require(func.boundInputLocations.length == aggAndReplaces.length,
+          s"Input locations for ${func.windowFunc} do not match aggregations " +
+              s"${func.boundInputLocations.toSeq} vs $aggAndReplaces")
+        val combined = withResource(
+          new ArrayBuffer[cudf.ColumnVector](aggAndReplaces.length)) { replacedCols =>
+          func.boundInputLocations.indices.foreach { aggIndex =>
+            val inputColIndex = func.boundInputLocations(aggIndex)
+            val inputCol = inputCb.column(inputColIndex).asInstanceOf[GpuColumnVector].getBase
+            val anr = aggAndReplaces(aggIndex)
+            val agg = anr.agg
+            val replacePolicy = anr.nullReplacePolicy
+            replacedCols +=
+                withResource(inputCol.scan(agg, ScanType.INCLUSIVE, NullPolicy.EXCLUDE)) {
+                  scanned =>
+                    // For scans when nulls are excluded then each input row that has a null in it
+                    // the output row also has a null in it. Typically this is not what we want,
+                    // because for windows that only happens if the first values are nulls. So we
+                    // will then call replace nulls as needed to fix that up. Typically the
+                    // replacement policy is preceding.
+                    if (replacePolicy.isDefined) {
+                      scanned.replaceNulls(replacePolicy.get)
+                    } else {
+                      scanned.incRefCount()
+                    }
```

Comment:
Can simplify this with something like:
```suggestion
                    replacePolicy.map(scanned.replaceNulls).getOrElse(scanned.incRefCount())
```